src/Umbraco.Core/Services/Implement/ContentService.cs

```diff
@@ -868,6 +868,10 @@ public PublishResult SaveAndPublish(IContent content, string culture = "*", int
                 if (!culture.IsNullOrWhiteSpace() && culture != "*")
                     throw new NotSupportedException($"Culture \"{culture}\" is not supported by invariant content types.");
             }
+            if (content.Name.Length > 255)
```

Comment:
I'd rather have this validation done at the Model/Controller level and throw an exception here instead. The validation can easily be done with data annotations on the `ContentVariantSave.Name` (This will be diff in v7!) property with a max length attribute. If you set that, then there's probably nothing else that needs to be done apart from throwing an exception here.

Reply:
@Shazwazza isnt it better to show a notification saying "too long name" rather than an exception like what its doing now? I ll certainly add the Maxlength attribute to the model. Donno why I didnt think of it at all!

Reply:
If the data annotations is added to the view model , it will be validated by the content controller and a nice message will be returned in the UI. If however someone is using the APIs directly at the service level, an exception is better. The service level doesn't return model validation results and we don't really want to start doing that. At the service level, an exception is preferred to tell the developer that they need to validate this before sending it to the service.

Reply:
Ahh. I didn’t think about the API users at all. I was only thinking of nice error messages to the editor in the CMS. I ll amend it